google/resource_google_project_default_service_accounts.go

```diff
@@ -0,0 +1,151 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/iam/v1"
+)
+
+// resourceGoogleProjectDefaultServiceAccounts returns a *schema.Resource that allows a customer
+// to manage all the default serviceAccounts.
+// It does mean that terraform tried to perform the action in the SA at some point but does not ensure that
+// all defaults serviceAccounts where managed. Eg.: API was activated after project creation.
+func resourceGoogleProjectDefaultServiceAccounts() *schema.Resource {
+	return &schema.Resource{
+		SchemaVersion: 0,
+
+		Create: resourceGoogleProjectDefaultServiceAccountsCreate,
+		Read:   resourceGoogleProjectDefaultServiceAccountsReadAndUpdate,
+		Update: resourceGoogleProjectDefaultServiceAccountsReadAndUpdate,
```

Comment:
The Update is only for one value that's not stored on the API. Meaning, it's just a state update. Maybe when I create all logics I'll have to split them, but so far I know it may sound odd, but it's behaving correctly.
```
carvalho-c02ccan9md6m:pdsa thiagocarvalho$ terraform destroy
google_project_default_service_accounts.name: Refreshing state... [id=projects/arc-seed-project]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # google_project_default_service_accounts.name will be destroyed
  - resource "google_project_default_service_accounts" "name" {
      - action         = "delete" -> null
      - id             = "projects/arc-seed-project" -> null
      - project_id     = "arc-seed-project" -> null
      - restore_policy = "REACTIVATE" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

google_project_default_service_accounts.name: Destroying... [id=projects/arc-seed-project]
google_project_default_service_accounts.name: Destruction complete after 0s

Destroy complete! Resources: 1 destroyed.
carvalho-c02ccan9md6m:pdsa thiagocarvalho$ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_project_default_service_accounts.name will be created
  + resource "google_project_default_service_accounts" "name" {
      + action         = "delete"
      + id             = (known after apply)
      + project_id     = "arc-seed-project"
      + restore_policy = "REACTIVATE"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_project_default_service_accounts.name: Creating...
google_project_default_service_accounts.name: Creation complete after 2s [id=projects/arc-seed-project]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
carvalho-c02ccan9md6m:pdsa thiagocarvalho$ vim terraform.tf
carvalho-c02ccan9md6m:pdsa thiagocarvalho$ terraform apply
google_project_default_service_accounts.name: Refreshing state... [id=projects/arc-seed-project]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_project_default_service_accounts.name will be updated in-place
  ~ resource "google_project_default_service_accounts" "name" {
        action         = "delete"
        id             = "projects/arc-seed-project"
        project_id     = "arc-seed-project"
      ~ restore_policy = "REACTIVATE" -> "NONE"
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_project_default_service_accounts.name: Modifying... [id=projects/arc-seed-project]
google_project_default_service_accounts.name: Modifications complete after 1s [id=projects/arc-seed-project]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
carvalho-c02ccan9md6m:pdsa thiagocarvalho$ 
```

Reply:
I could create another function for the update which just set the new state... I'll keep that in mind for the next updates